kompari/src/fsutils.rs

```diff
@@ -26,8 +27,9 @@ pub fn list_image_dir(dir_path: &Path) -> Result<impl Iterator<Item = PathBuf>,
 
 pub fn list_image_dir_names(
     dir_path: &Path,
-) -> Result<impl Iterator<Item = OsString>, std::io::Error> {
-    Ok(list_image_dir(dir_path)?.filter_map(|p| p.file_name().map(|name| name.to_os_string())))
+) -> Result<impl Iterator<Item = PathBuf>, std::io::Error> {
+    let dir_path = dir_path.to_path_buf();
```

Comment:
I think that this method should either accept a `PathBuf` itself, or mark the returned iterator as `'_` (i.e. borrowing the input path, which I believe that you can do)